### PR TITLE
Add ability to create scopes and collections

### PR DESCRIPTION
**Pull request by jgroh9**
# What was done

- Added a new script to create scopes and collections for Couchbase 7
- Added documentation to the readme on configuring scopes and collections
- Added sample configuration files for the scope/collection creation
- Tested new logic locally via modifying the Dockerfile and docker-compose file in the example/sample directory

### CODE REVIEW

**Review by brantburnett — CHANGES_REQUESTED** [review]
First, let me say thanks so much for the submission. Scopes/collections are something we really need to get added, and I appreciate the assistance.

My one concern here is the way you're approaching fakeit. I agree that it's an effective approach for a single scope/collection, given the current limitations of fakeit. My worry is that, once we've added it working this way, we'll have to support this approach indefinitely for backward compatibility. It will be more than a little cumbersome once fakeit is up to snuff, especially since it forces the same scope/collection name for all buckets.

I'd like to propose that we pare this PR down a bit to just create the scopes/collections (for now), get that merged, and then work separately on how to make fakeit work better with scopes/collections.

**Comment by jgroh9** [comment]
Ok, I will make that change

**Comment by jgroh9** [comment]
Changes to remove FakeIt scope/collection data logic have been removed

File: scripts/create-collections.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  echo "Creating scopes and collections in $bucketName..."
+
+  # Create the scopes and collections
+  while read scopeName
+  do
+    echo "Scope: $scopeName"
+    # Check to see if the scope name provided is the default scope. If not, create the scope
+    if [ $scopeName != "_default" ]; then
+      echo "Creating scope: $scopeName..."
+      curl -Ss -X POST -u "$CB_USERNAME:$CB_PASSWORD" http://localhost:8091/pools/default/buckets/$bucketName/collections -d name=$scopeName
```
Comment (brantburnett) [inline scripts/create-collections.sh:15]: This ensures a newline in the script output

```suggestion
      curl -Ss -X POST -u "$CB_USERNAME:$CB_PASSWORD" http://localhost:8091/pools/default/buckets/$bucketName/collections -d name=$scopeName && echo
```

File: scripts/create-collections.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+      sleep 5
```
Comment (brantburnett) [inline scripts/create-collections.sh:17]: The sleep seems unnecessary in my testing

File: scripts/create-collections.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    fi
+
+    # Create the collections within the appropriate scope
+    while read collectionName
+    do
+      echo "Creating collection: $collectionName..."
+      curl -Ss -X POST -u "$CB_USERNAME:$CB_PASSWORD" http://localhost:8091/pools/default/buckets/$bucketName/collections/$scopeName -d name=$collectionName
```
Comment (brantburnett) [inline scripts/create-collections.sh:24]: This ensures a newline in the script output

```suggestion
      curl -Ss -X POST -u "$CB_USERNAME:$CB_PASSWORD" http://localhost:8091/pools/default/buckets/$bucketName/collections/$scopeName -d name=$collectionName && echo
```

**Review by brantburnett — CHANGES_REQUESTED** [review]
Just a few more details, looks great!

**Comment by jgroh9** [comment]
Requested changes made

**Review by brantburnett — APPROVED** [review]

**Pull request merged**